Automattic/wp-calypso
# Launchpad: Replace my home url with redirect to launchpad

## Context

As part of the launchpad initiative, we'd like to ensure that launchpad enabled sites show the LaunchPad instead of calypso "My Home". To address this, we modify the Calypso `page` client side router to redirect from myhome to launchpad when appropriate.

![2022-08-16 12 07 08](https://user-images.githubusercontent.com/5414230/184962074-c83dd90c-2f79-4d61-bbbe-f3d6aac2f840.gif)

## Proposed Changes

* Set up logic to redirect users to launchpad if they visit my home ( won't be complete until https://github.com/Automattic/wp-calypso/issues/66706 and https://github.com/Automattic/wp-calypso/issues/66707 are implemented )
* Gate redirect behind launchpad feature flag

## Testing Instructions


* Sandbox a site and public API with this phab diff D86659-code
* Initiate SSH connection to sandbox and update site option for a test site
  * Execute `wpsh` in ssh'd terminal
  * `return switch_to_blog(YOUR_SITE_ID_HERE)`
  * `return update_option('launchpad_screen', 'full')`
  * `return update_option('site_intent', 'link-in-bio')`
  * exit the `wpsh` shell
* Check out this branch
* `yarn start`
* Attempt to navigate to my home for _any_ site, either through the UI or directly inputting a `/home` url into the address bar Ex. http://calypso.localhost:3000/home/YOUR_SITE_SLUG_HERE
* Note that you are redirected to the link-in-bio launchpad flow
* Update the launchpad_screen option to 'off'
* Note that you are no longer redirected to the launchpad flow

When https://github.com/Automattic/wp-calypso/issues/66706 and https://github.com/Automattic/wp-calypso/issues/66707 are complete, users will only be redirected if launchpad is enabled _and_ their checklist tasks **have not** been completed.

## Pre-merge Checklist






Related to https://github.com/Automattic/wp-calypso/issues/66573
